### PR TITLE
Reverse the included relation name formatting

### PR DESCRIPTION
**Pull request by schtibe**
See https://github.com/django-json-api/django-rest-framework-json-api/issues/203 for more information

### CODE REVIEW

**Comment by schtibe** [comment]
I have yet to figure out how the unit testing works...

**Comment by jerel** [comment]
As far as the tests go `python runtests.py` should get you up and running. Is there something specific you're wondering about? You can add a function to this file https://github.com/django-json-api/django-rest-framework-json-api/blob/develop/example/tests/integration/test_includes.py and it should just be ran with the tests. Name it similar to the ones in there and if you add `single_entry` as function parameters pytest will know to create mocks for you.

**Comment by schtibe** [comment]
Right now I have problems like this http://paste.codedump.ch/view/837a4192 (regardless of the repos)

**Comment by schtibe** [comment]
I think for it to test the existing code I'd have to rename all the 'comments' and 'authors' to singular.. which has an effect on the "relationships" too.. not sure if it's OK to do that?

File: example/tests/integration/test_includes.py [hunk 1]
```diff
@@ -50,3 +50,21 @@ def test_missing_field_not_included(author_bio_factory, author_factory, client):
...
     assert data['included'][0]['attributes']['body'] == bio.body
+
+def test_reverse_included(single_entry, client):
+    """Test the parsing of included names"""
+    from django.conf import settings
+
+    parse_relation       = getattr(settings, 'JSON_API_PARSE_INCLUDE_KEYS', None)
+    singularize_included = getattr(settings, 'JSON_API_SINGULARIZE_INCLUDE_TYPE', None)
+
+    settings.JSON_API_PARSE_INCLUDE_KEYS = 'underscore'
```
Comment (jesushernandez) [inline example/tests/integration/test_includes.py]: pytest-django provides a good way of doing this: http://pytest-django.readthedocs.org/en/latest/helpers.html#settings

**Comment by birkholz** [comment]
Quote from #257:
> It seems that #204 is taking things further by also reversing pluralization and adding a setting to specify which format the keys should be reverted to. In my opinion, the effort of that PR should be expanded to the other areas that perform this behavior, namely parsing. This could then be considered a stopgap until that larger development is ready.

I suggest this PR be expanded to also apply the reversal to attribute/relationship parsing [here](https://github.com/django-json-api/django-rest-framework-json-api/blob/develop/rest_framework_json_api/parsers.py#L32).

**Comment by mblayman** [comment]
Hi, @schtibe. Thanks for the contribution! This branch is now in conflict in the main branch. Are you interested in fixing this up or should we close out this PR? I'm stepping up to maintain the PR queue and get DRF JA changes flowing again, but I can't merge conflicted stuff.

If this is still valuable, resolving the merge conflicts will give me a chance to review it well. If this is no longer important to you, closing the PR is right next step.

Please let me know what you'd like to do. Thanks!

**Comment by schtibe** [comment]
Hi @mblayman. I'd like to see if I can fix the the conflict, but I'm currently not even sure anymore if the issue adressed in this PR really complies with the JSON API standard. Please give me some time as I cannot check it right away.

**Comment by schtibe** [comment]
Ok I don't see the possibility any time soon to investigate further.

**Comment by auvipy** [comment]
will try to investigate this in near future

**Pull request closed without merging**